keyboards/aidansmithdotdev/fine40/keymaps/default/keymap.c

```diff
@@ -0,0 +1,112 @@
+        0,  0,  0,  0,  0,224,240,240,240,192,128,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,128,224,240,240,240,224,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,252,252,252,252,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,120,248,248,112,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,128,192,224,240,240,240,224,  0,  0,  0,  0,  0,  0,128,192,224,240,240,240,240,240,240,224,224,192,  0,  0,  0,  0,  0,
+        0,  0,  0,  0,224,255,255,255, 15, 63,255,254,248,224,  0,  0,  0,  0,224,248,254,255, 31, 31,255,255,255,128,  0,  0,  0,128,224,240,248,252, 60, 28, 28, 30, 28, 60,124,248,248,240,192,  0,  0,  0,128,224,240,248,252, 60, 60, 28, 30, 28, 28, 60, 28,  0,  0,  0,255,255,255,255, 28, 28, 28, 28, 60,124,252,248,240,192,  0,  0,  0,  0,252,252,252,252,  0,  0,  0,  0,  0,192,224,248,252, 63, 31,  7,  3,255,255,255,255,  0,  0,  0,  0,224,254,255,255, 15,  1,  0,  0,  0,  0,  1, 15,255,255,255,248,  0,  0,  0, 
+        0,  0,  0,  0,255,255,255,255,  0,  0,  0,  3, 15, 63,255,252,252,255, 63, 15,  3,  0,  0,  0,255,255,255,255,  0,  0,  0, 63,255,255,255,224,128,  0,  0,  0,  0,128,192,255,255,255,127,  4,  0,  0, 63,255,255,255,224,128,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,255,255,255,255,  0,  0,  0,  0,  0,  0,255,255,255,255,  0,  0,  0,  0,255,255,255,255,  0,  0,  0, 60, 63, 63, 63, 61, 60, 60, 60, 60, 60,255,255,255,255, 60, 60,  0,  0,  7,127,255,255,248,128,  0,  0,  0,  0,128,240,255,255,127, 31,  0,  0,  0, 
+        0,  0,  0,  0, 15, 15, 15,  7,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  7, 15, 15, 15,  0,  0,  0,  0,  0,  1,  3,  7,  7, 15, 15, 15, 15, 15,  7,  7,  3,  1,  0,  0,  0,  0,  0,  0,  3,  3,  7,  7, 15, 15, 15, 15, 15, 15,  7,  0,  0,  0, 15, 15, 15,  7,  0,  0,  0,  0,  0,  0,  7, 15, 15, 15,  0,  0,  0,  0, 15, 15, 15,  7,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0,  0, 15, 15, 15,  7,  0,  0,  0,  0,  0,  0,  1,  3,  7,  7, 15, 15, 15, 15,  7,  7,  3,  1,  0,  0,  0,  0,  0,
+    };
+	
+	oled_write_raw_P(mochi_logo, sizeof(mochi_logo));
+}
+
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+       return OLED_ROTATION_180;  // flips the display 180 degrees
+}
+bool oled_task_user(void) {
+	render_mochi();
+	oled_set_cursor(0, 4);
+	oled_write_P(PSTR("Layer: "), false);
+	
+	switch(get_highest_layer(layer_state))
+	{
+		case _MAIN:
+			oled_write_ln_P(PSTR("Main"), false);
+			break;
+		case _LEFT:
+			oled_write_ln_P(PSTR("Left"), false);
+			break;
+		case _RIGHT:
+			oled_write_ln_P(PSTR("Right"), false);
+			break;
+		case _TAB:
+			oled_write_ln_P(PSTR("Tab"), false);
+			break;
+	}
+	
+	//render_logo();
+	return(true);
+}
+#endif
+
+#ifdef ENCODER_ENABLE
+bool encoder_update_user(uint8_t index, bool clockwise) {
+	// Volume control
+	if (clockwise) {
+		tap_code(KC_VOLU);
+	} else {
+		tap_code(KC_VOLD);
+	}
+	return false;
+}
+#endif
```

Comment:
This should be moved to a `fine40.c`, and use the `*_kb` versions of the function, instead.  This way, you can specify a default behavior once, and it will be used for the configurator as well. 
```suggestion
```

Reply:
Do you have any guide for that conversion? From your explanation I put "_kb" after all of the QMK functions and QMK says those functions do not exist.

Reply:
@drashna Could you provide a more detailed explanation or an example of this being done? I can't find it in any documentation.

Reply:
Nevermind, I finally found what I was doing wrong. An important package had not been included. A push up with the correct code is inbound.